src/main/scala/dpla/ingestion3/mappers/providers/NaraExtractor.scala

```diff
@@ -0,0 +1,256 @@
+    DplaMapData(
+      DplaSourceResource(
+        collection = collection(xml).map(nameOnlyCollection),
+        contributor = contributor(xml).map(nameOnlyAgent),
+        creator = creator(xml).map(nameOnlyAgent),
+        date = date(xml),
+        description = extractStrings("scopeAndContentNote"),
+        extent = extractStrings("extent"),
+        format = extractStrings(xml \\ "specificRecordsTypeArray" \ "specificRecordsType" \ "termName"),
+        identifier = extractStrings("naId"),
+        language = extractStrings(xml \\ "languageArray" \ "language" \ "termName").map(nameOnlyConcept),
+        place =
+          extractStrings(xml \\ "geographicReferenceArray" \ "geographicPlaceName" \ "termName").map(nameOnlyPlace),
+        publisher = publisher(xml).map(nameOnlyAgent),
+        relation = relation(xml).map(Left(_)),
+        rights = rights(xml),
+        subject = extractStrings(xml \\ "topicalSubjectArray" \ "topicalSubject" \ "termName").map(nameOnlyConcept),
+        title = extractStrings("title"),
+        `type` = types(xml)
+      ),
+      EdmWebResource(
+        //todo proper uri handling from other branch
+        uri = itemUri(xml),
+        //TODO naId doesn't seem like a meaningful file format
+        fileFormat = extractStrings(xml \\ "objectType" \ "naId") ++ extractStrings(xml \\ "objectType" \ "termName")
+      ),
+      OreAggregation(
+        uri = itemUri(xml), //this should probably be renamed to isShownAt
+        dataProvider = dataProvider(xml),
+        originalRecord = rawData,
+        provider = agent,
+        preview = extractString("thumbnailFilename").map(new URI(_)).map(uriOnlyWebResource)
+      )
+    )
+  }
+
+  def agent = EdmAgent(
+    name = Some("National Archives and Records Administration"),
+    uri = Some(new URI("http://dp.la/api/contributor/nara"))
+  )
+
+  def collection(xml: NodeSeq): Seq[String] = {
+    val parentRecordGroupIds = for {
+      prg <- xml \ "parentRecordGroup"
+      prgId <- prg \ "naId" :: prg \ "title" :: prg \ "recordGroupNumber" :: Nil
+    } yield prgId.text
+
+    val parentCollectionIds = for {
+      pc <- xml \ "parentCollection"
+      pcId <- pc \ "naId" :: pc \ "title" :: Nil
+    } yield pcId.text
+
+    if (parentRecordGroupIds.nonEmpty) parentRecordGroupIds else parentCollectionIds
+  }
+
+  def contributor(xml: NodeSeq): Seq[String] = {
+
+    //TODO: not handling multiple display values. haven't found example yet.
+
+    val organizationalContributors = for {
+      org <- xml \\ "organizationalContributorArray" \ "organizationalContributor"
+      name = (org \ "contributor" \ "termName").text
+      _type = (org \ "contributorType" \ "termName").text
+      if _type != "Publisher"
+    } yield name
+
+    val personalContributors = for {
+      person <- xml \\ "personalContributorArray" \ "personalContributor"
+      name = (person \ "contributor" \ "termName").text
+      //_type = (person \ "contributorType" \ "TermName").text
```

Comment:
It's mentioned in the spec, but doesn't do anything. Not quite sure what to do with it, to be honest!